src/redis-cli.c

```diff
@@ -992,6 +995,8 @@ static int parseOptions(int argc, char **argv) {
             config.stdinarg = 1;
         } else if (!strcmp(argv[i],"-p") && !lastarg) {
             config.hostport = atoi(argv[++i]);
+        } else if (!strcmp(argv[i],"-t") && !lastarg) {
+            config.connect_timeout = (struct timeval){.tv_sec=atoi(argv[++i])};
```

Comment:
I think it's wiser to allow specifying the timeout in milliseconds. Same thing about casting, should not be needed AFAIK. About milliseconds, IMHO it's ok to specify seconds but as float as we do in other instances in redis-cli: it will get converted into a double, that can be converted into the number of seconds / milliseconds. It's very useful to way, for instance, 100 ms or 10 ms, because often 1 second is too much depending on the use case.